api/query/query.go

```diff
@@ -61,6 +57,27 @@ func (qh queryHandler) processInput(w http.ResponseWriter, r *http.Request) (*sh
 	return &filters, testRuns, summaries, nil
 }
 
+func (qh queryHandler) validateSummaryVersions(v url.Values, logger shared.Logger) (bool, error) {
```

Comment:
You can return an error here instead of a boolean and error. This will simplify the signature of the method. For the most part every time you return false, you also return an error. The only major change would be at line 75. You should return a custom error.

More context around the change: From the caller's perspective, a method named "validate[SomeName]", it makes sense to return an error to indicate success or failure. But with a boolean and an error, there's an extra step the caller needs to do to ensure it really was validated.

Reply:
I see, and I agree with you. I made the change to the best of my knowledge (hopefully it's understandable).